test/DynamoCoreTests/AuthenticationManagerTests.cs

```diff
@@ -47,5 +57,56 @@ public void Login_CausesLoginMethodToBeInvokedOnAuthProvider()
         }
 
         #endregion
+
+        #region Toggle Login State
+
+        [Test]
+        public void ToggleLoginState()
+        {
+            bool loginCalled = false;
+
+            var authProviderMock = new Mock<IAuthProvider>();
+            authProviderMock.Setup(x => x.Login()).Callback(() => loginCalled = true);
+
+            var authManager = new AuthenticationManager(authProviderMock.Object);
+
+            authManager.ToggleLoginState(authManager);
+            Assert.True(loginCalled);
+        }
+
+        [Test]
+        public void CanToggleLoginState()
+        {
+            var authProviderMock = new Mock<IAuthProvider>();
+            var authManager = new AuthenticationManager(authProviderMock.Object);
+
+            Assert.IsTrue(authManager.CanToggleLoginState(authManager));
+        }
+
+        #endregion
+
+        #region Properties
+
+        [Test]
+        public void Property_Username()
+        {
+            var authProviderMock = new Mock<IAuthProvider>();
+            var authManager = new AuthenticationManager(authProviderMock.Object);
+
+            authProviderMock.Setup(x => x.Username).Returns("test");
+
+            Assert.AreEqual(authManager.Username, "test");
```

Comment:
Would you also assert the other public members here? e.g. `HasAuthProvider`

Reply:
The other member appear as covered. Maybe by some other tests.

Reply:
@QilongTang Should I add specific tests for those members, even though they are already covered by the current tests? (For example, HasAuthProvider is covered when testing the Username property)
![image](https://user-images.githubusercontent.com/34453173/77569190-6a3d4380-6e8f-11ea-92b3-5793337c586b.png)

Reply:
Sorry not meant for asking to create new tests. If you can cover asserting `HasAuthProvider` and `LoginState`, that's good enough

Reply:
Two tests were added to check for the HasAuthProvider and LoginState properties.